src/components/Asset/AssetActions/Compute/_constants.ts

```diff
@@ -1,13 +1,51 @@
+import { FormConsumerParameter } from '@components/Publish/_types'
+import { parseConsumerParameters } from '@utils/ddo'
 import * as Yup from 'yup'
 
+export const ConsumerParametersConsumeSchema = Yup.array()
+  .of(
+    Yup.object().shape({
+      required: Yup.boolean().required('required'),
+      default: Yup.mixed().when('required', {
+        is: false,
+        then: Yup.mixed()
+          .nullable()
+          .transform((value) => value || null),
+        otherwise: Yup.mixed().required('Required')
+      })
+    })
+  )
+  .nullable()
+
 export const validationSchema: Yup.SchemaOf<{
   algorithm: string
+  dataService: any
+  algoService: any
```

Comment:
suggestion: these could probably be renamed into `dataServiceParams` and `algoServiceParams` respectively to better communicate what they are used for. (Should be adapted to different relevant sections in the code)